### PR TITLE
Expanding augmentation lambda README

### LINKED ISSUE #528: Create documentation/readme for how TTC is Augmenting eICRS
Leveraging different pieces of documentation, create a README for how augmentation works in TTC.

Diagram:
https://drive.google.com/file/d/1yhq3xDRFemau6hVECXQxCIw3oFP2nAok/view

Diagram Output:
https://drive.google.com/file/d/1EAxRvgSUVxfSupNvvdT_Fi6-tkSIueRp/view

Augmentation process (mostly correct, but with some missing recent changes):

https://docs.google.com/document/d/1dZSWvP3GWleU_JjE3w5IqxXBCqublGKB0xwf6yb5ycU/edit?tab=t.0#heading=h.w6vn8ufyqkpw

### PR DESCRIPTION
**Pull request by robertandremitchell**
## Description

This expands the augmentation-lambda documentation to include more detail, based on writeups and diagrams here: https://app.zenhub.com/workspaces/dibbs-text-to-code-68756bf8ea06bb00319391c0/issues/gh/cdcgov/dibbs-text-to-code/528.

I'm not sure if we want to expand, or take a step back and add more of this detail to the overall README to keep augmentation-lambda focused on code implementation. I tend to prefer overall README's to be the more overview, but I can see an argument since all these lambdas are intended to run together to have more of the detail on how Index->TTC->Augmentation work somewhere else. Anyway.

## Related Issues

Closes #528 

## Additional Notes

[Add any additional context or notes that reviewers should know about.]

<--------------------- REMOVE THE LINES BELOW BEFORE MERGING --------------------->

## Checklist

Please review and complete the following checklist before submitting your pull request:

- [ ] I have ensured that the pull request is of a manageable size, allowing it to be reviewed within a single session.
- [ ] I have reviewed my changes to ensure they are clear, concise, and well-documented.
- [ ] I have updated the documentation, if applicable.
- [ ] I have added or updated test cases to cover my changes, if applicable.
- [ ] I have minimized the number of reviewers to include only those essential for the review.

## Checklist for Reviewers

Please review and complete the following checklist during the review process:

- [ ] The code follows best practices and conventions.
- [ ] The changes implement the desired functionality or fix the reported issue.
- [ ] The tests cover the new changes and pass successfully.
- [ ] Any potential edge cases or error scenarios have been considered.

### CODE REVIEW

File: packages/augmentation-lambda/README.md [hunk 1]
```diff
@@ -17,6 +17,10 @@ Augmentation, specifically for the work in this repo, is related to eCR messages
 
 The Lambda package is intentionally kept separate from the core augmentation package. The core `augmentation` package owns the reusable XML augmentation behavior, while `augmentation-lambda` owns the AWS event handling, S3 reads and writes, and Lambda-specific orchestration.
```
Comment (BradySkylight) [inline packages/augmentation-lambda/README.md:18]: this text may need to go away now, as I don't think the plan is to have a centralized augmenter between the various components anymore...though we may still need APHL to confirm this.

File: packages/augmentation-lambda/README.md [hunk 2]
```diff
@@ -35,13 +39,380 @@ This package is responsible for:
...
+- Unmatched schematron errors grouped by data field.
+
+Matched schematron errors that include `new_translation` are converted into `NonstandardCodeInstance` objects and passed into the core augmentation package.
+
+The current TTC output shape is:
+
+```json
+{
+  "persistence_id": "<persistence id>",
+  "eicr_metadata": {},
```
Comment (nickclyde) [inline packages/augmentation-lambda/README.md]: Small accuracy nit: I think `eicr_metadata` is a bit misleading as `{}` here. Looking at the TTC Lambda outputs, when populated it ends up as a string like `"eicr_id=CdaInstanceIdentifier(...) eicr_vendor='...'"` rather than a JSON object. Could we add an example of the real expected shape of the metadata, eg: "eicr_id=CdaInstanceIdentifier(null_flavor=None, assigning_authority_name=None, displayable=None, root='a2e68734-045a-4136-b8f4-1bbf1fcd4e2e', extension=None) eicr_vendor='Test eCR Vendor Name'"

File: packages/augmentation-lambda/README.md [hunk 3]
```diff
@@ -35,13 +39,380 @@ This package is responsible for:
...
+The original code element is not replaced. Augmentation preserves the original source content and appends the TTC-generated mapped code as a translation.
+
+The XML translation uses CDA-style XML attributes:
+
+```xml
+<translation
+  code="<code>"
+  codeSystem="2.16.840.1.113883.6.1"
+  codeSystemName="LOINC"
+  DisplayName="<display name>"
```
Comment (nickclyde) [inline packages/augmentation-lambda/README.md]: Should this be `displayName` instead of `DisplayName`?

**Review by nickclyde — APPROVED** [review]
Awesome work Rob!!

File: packages/augmentation-lambda/README.md (hunk 1 above)
Comment (robertandremitchell) [reply packages/augmentation-lambda/README.md:18]: good call. updating!

File: packages/augmentation-lambda/README.md (hunk 2 above)
Comment (robertandremitchell) [reply packages/augmentation-lambda/README.md]: makes sense, adding
